### DISCUSSION TITLE
How about improving the SuspenseQuery Hooks to display an error through type checking when using `skipToken`?

### DISCUSSION THREAD

**Discussion by gwansikk** [TanStack/query#8078] [Ideas] [answered]
I wanted to use `SkipToken` in the SuspenseQuery Hooks, but I found out through the developer console that `SkipToken` cannot be used in SuspenseQuery.

```ts
 if (process.env.NODE_ENV !== 'production') {
    if (options.queryFn === skipToken) {
      console.error('skipToken is not allowed for useSuspenseInfiniteQuery')
    }
  }
```

Is there a reason for notifying developers via console logs instead of TypeScript type checking?

I believe that showing this through browser console logs during development does not provide a good DX(developer experience). How about presenting it as a TypeScript type check error so developers can be aware of it during development?

**Comment by TkDodo** [comment, marked as answer]
yeah should be as easy as `queryFn: Exclude<options['queryFn'], SkipToken>` but not sure. Do you want to give it a try to contribute it?

**Reply by gwansikk** [reply]
Sure, I'll give it a try. I have a good idea!
